portfolio/src/main/webapp/script.js

```diff
@@ -11,18 +11,71 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
+ 
+var currentIndex = -1;
 /**
- * Adds a random greeting to the page.
- */
+* Adds a random greeting to the page.
+*/
 function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+ const greetings =
+     ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+ 
+ // Pick a random greeting.
+ const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+ 
+ // Add it to the page.
+ const greetingContainer = document.getElementById('greeting-container');
+ greetingContainer.innerText = greeting;
+}
+ 
+function nextPicture(prefix){
+   currentIndex++;
```

Comment:
You could use a modulus operator (%) here, so the number can wrap around.
`currentIndex = (currentIndex + 1) % numPictures`

This way you won't need to check if `currentIndex < 0`

Reply:
Same for prevPicture